sdk/identity/identity/src/credentials/managedIdentityCredential/appServiceMsi2017.ts

```diff
@@ -25,21 +28,26 @@ function prepareRequestOptions(resource: string, clientId?: string): RequestPrep
     queryParameters.clientid = clientId;
   }
 
+  const query = qs.stringify(queryParameters);
+
   return {
-    url: process.env.MSI_ENDPOINT,
+    url: `${process.env.MSI_ENDPOINT!}?${query}`,
```

Comment:
Do we do any validation anywhere to see whether MSI_ENDPOINT is actually set to something in environment variables?